### PR TITLE
Reload memberAssignment

### PR DESCRIPTION
**Pull request by amberwebb**

### CODE REVIEW

**Comment by nathanielvelez** [comment]
👍

File: src/persistence.coffee [hunk 1]
```diff
@@ -754,6 +754,12 @@ modifySDK = (sdk) ->
         sdk.loadAssignments({id: memberAssignment.assignmentId}).then -> result
         ).callback callback
 
+  wrapMethod sdk, 'saveAssignment', (saveAssignment) ->
+    (assignment, callback) ->
+      saveAssignment.call(this, assignment, callback).then((result) ->
+        sdk.loadMemberAssignments({id: result.id})
+        ).callback callback
```
Comment (matthopson) [inline src/persistence.coffee]: indent?

File: src/persistence.coffee [hunk 2]
```diff
@@ -754,6 +754,12 @@ modifySDK = (sdk) ->
         sdk.loadAssignments({id: memberAssignment.assignmentId}).then -> result
         ).callback callback
 
+  wrapMethod sdk, 'saveAssignment', (saveAssignment) ->
```
Comment (matthopson) [inline src/persistence.coffee]: What about deleting?

Comment (matthopson) [inline src/persistence.coffee]: Also, we will probably need to ensure `memberAssignments` get removed when deleting an `event`

File: src/persistence.coffee [hunk 3]
```diff
@@ -754,6 +754,12 @@ modifySDK = (sdk) ->
...
+    (assignment, callback) ->
+      saveAssignment.call(this, assignment, callback).then((result) ->
+        sdk.loadMemberAssignments({id: result.id})
```
Comment (matthopson) [inline src/persistence.coffee]: This should be `assignmentId`

File: src/persistence.coffee (hunk 2 above)
Comment (amberwebb) [inline src/persistence.coffee]: What is there to load if an assignment is deleted than it's member assignment is deleted as well right?

Comment (matthopson) [inline src/persistence.coffee]: Not in memory on the front end. That's why this persistence layer exists.

Load `memberAssignments` now. Delete the associated `event`. You will likely see the same `memberAssignments.length` before and after deleting the `event`.

**Comment by amberwebb** [comment]
@matthopson Please review again. I am still having an issue with unlinking the member from assignments when assignments.memberId is null.

File: src/persistence.coffee [hunk 4]
```diff
@@ -740,19 +747,41 @@ modifySDK = (sdk) ->
...
+
+  wrapMethod sdk, 'deleteAssignment', (deleteAssignment) ->
+    (assignment, callback) ->
+      if assignment.memberAssignments.length
+        toRemove = []
+        assignment.memberAssignments.forEach (memberAssignment) ->
+          toRemove.push memberAssignment.member...
+          toRemove.push memberAssignment.event...
+          toRemove.push memberAssignment.assignment...
+        linking.unlinkItems toRemove, lookup
```
Comment (matthopson) [inline src/persistence.coffee]: When deleting an assignment, you should only have to unlink related `memberAssignment` objects - as they should be cleaned up for you by v3 when the `assignment` goes away. cc @jjmocko

**Comment by amberwebb** [comment]
@matthopson Pushed changes. Please review.

Setting result.member = null did not work because the result return was for memberAssignment. I need the member on assignment to be null, or ideally unlinked.

**Comment by matthopson** [comment]
Oh durr... 😛

**Comment by matthopson** [comment]
👍

**Pull request merged**